clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -1100,6 +1103,26 @@ void SYCLToolChain::TranslateGPUTargetOpt(const llvm::opt::ArgList &Args,
   }
 }
 
+static void WarnForDeprecatedBackendOpts(const Driver &D,
+                                         const llvm::Triple &Triple,
+                                         StringRef Device, StringRef ArgString,
+                                         const llvm::opt::Arg *A) {
+  // Suggest users passing GRF backend opts on PVC to use
+  // -ftarget-register-alloc-mode and
+
+  if (!ArgString.contains("-device pvc") && !Device.contains("pvc"))
+    return;
+  // Make sure to only warn for once for gen targets as the translate
+  // options tree is called twice but only the second time has the
+  // device set.
+  if (Triple.isSPIR() && Triple.getSubArch() == llvm::Triple::SPIRSubArch_gen &&
```

Comment:
this would probably need some kind of cleanup so we can do BE arg checking from anywhere and not be impacted by multiple diagnostic outputs, but yeah, it's not pretty.